src/app_service/service/wallet_account/service.nim

```diff
@@ -106,11 +117,12 @@ QtObject:
     accountsService: accounts_service.Service
     tokenService: token_service.Service
     networkService: network_service.Service
-    walletAccounts: OrderedTable[string, WalletAccountDto]
     processedKeyPair: KeyPairDto
+    walletAccountsLock: Lock
```

Comment:
Why do we need that? Is any async task operating directly on `walletAccounts`?

Reply:
Crash I was facing is when multiple wallet accounts were added in a row (creating new keycard (new/old seed phrase) or importing a keypair from a keycard), when it was like few ms difference among each of them, for each added wallet account we're building tokens which is done in a separate thread, when responses come to each of those threads the same slot was called which operates above the same container (`OrderedTable`) and it looks that was done in the context of thread where response is received and I was getting a crash. After locking the table I haven't seen crash at all.

Reply:
From my understanding, slots are supposed to be invoked in the main (receiver) thread and because of that, we haven't needed any synchronization mechanism until now.

Maybe it needs a second look to determine what the real problem is. I could try to help with the debugging.

Reply:
@osmaczko yes, I was surprised as well, but locking the table fixed issues for me. My understanding was that the jobs are done in a separate thread, but slots are done in the main thread. Actually that is something completely unrealated to the subject of this issue and this PR, I just was facing it during my work here and wanted to sort it out. But anyhow would be great if you or anybody else can rever that commit (related changes) and try to reproduce a crash and have another look into that. May I can check that myself and spend some more time investigating there, but cannot promise that will be these days.

Here is an issue to keep that on mind: https://github.com/status-im/status-desktop/issues/9115